src/attributes/components/AttributeDetails/NumericUnits.tsx

```diff
@@ -28,46 +37,68 @@ const useStyles = makeStyles(
   { name: "NumericUnits" }
 );
 
-interface NumericUnitsProps {
-  data: AttributePageFormData;
+interface UnitData {
+  unit: MeasurementUnitsEnum;
+  system: UnitSystem;
+  type: UnitType;
+}
+
+interface NumericUnitsProps
+  extends Pick<
+    UseFormResult<AttributePageFormData>,
+    "set" | "setError" | "data" | "errors" | "clearErrors"
+  > {
   disabled: boolean;
-  error: string;
-  onChange: (event: React.ChangeEvent<any>) => void;
-  set: (data: Partial<AttributePageFormData>) => void;
-  triggerChange: () => void;
 }
 
 export const NumericUnits: React.FC<NumericUnitsProps> = ({
   data,
   disabled,
-  onChange,
+  errors,
   set,
-  triggerChange,
-  error
+  setError,
+  clearErrors
 }) => {
   const classes = useStyles();
-  const unitRef = useRef(data.unit);
-  const [system, setSystem] = useState(null);
-  const [type, setType] = useState(null);
+  const [unitData, setUnitData] = useState<Partial<UnitData>>({
+    unit: data.unit
+  });
+
+  const { unit, system, type } = unitData;
+  const errorProps = { error: !!errors.unit, hint: M.required };
+
+  useEffect(() => set({ unit }), [unit]);
 
   useEffect(() => {
     if (data.unit) {
+      const initialData = { unit: data.unit } as UnitData;
```

Comment:
Or maybe make it also like it is in state:
```suggestion
      const initialData: Partial<UnitData> = { unit: data.unit };
```